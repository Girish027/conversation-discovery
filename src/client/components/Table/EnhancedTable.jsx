import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table, Filter } from '@tfs/ui-components';
import IconButton from 'components/IconButton';

import { saveStorage, loadStorage } from '../../utils/storageManager';

export const getColumnWidth = (tableState = {}, columnId, defaultWidth) => {
  const columnState = tableState[columnId] || {};
  const { resizedWidth = defaultWidth } = columnState;
  return resizedWidth;
};

export const getColumnVisibility = (tableState = {}, columnId, defaultVisiblity) => {
  const columnState = tableState[columnId] || {};
  const { visible = defaultVisiblity } = columnState;
  return visible;
};

export default class EnhancedTable extends Component {
  constructor(props) {
    super(props);
    this.props = props;
    this.handleTableResizeChange = this.handleTableResizeChange.bind(this);
    this.handleColumnSelection = this.handleColumnSelection.bind(this);
  }

  handleTableResizeChange(newResized) {
    const { tableIndentifier } = this.props;
    let resizedData = {};
    let modifiedData = {};
    let storedTableState = {};

    if (newResized && tableIndentifier) {
      newResized.forEach((column) => {
        const resizedCol = column.id;
        const resizedWidth = column.value;

        resizedData = Object.assign(resizedData, {
          [resizedCol]: {
            activeCol: resizedCol,
            resizing: true,
            resizedWidth,
          },
        });

        storedTableState = loadStorage(tableIndentifier, {});

        if (storedTableState) {
          resizedData = Object.assign({}, {
            [resizedCol]: { ...storedTableState[resizedCol], ...resizedData[resizedCol] },
          });
        }
      });

      modifiedData = Object.assign({}, storedTableState, resizedData);
      // saveTableState(modifiedData, tableIndentifier);
      saveStorage(tableIndentifier, modifiedData);
    }

    return modifiedData;
  }

  // TODO: Column Selector Menu is yet to be implemented.
  // Below is the handler for it.
  handleColumnSelection(columnId) {
    const { tableIndentifier } = this.props;
    const storedTableState = loadStorage(tableIndentifier, {});
    const columnState = storedTableState[columnId] || {};
    const { visible: currentVisibleState = true } = columnState;

    const newColumnState = Object.assign({}, columnState, {
      [columnId]: {
        visible: !currentVisibleState,
      },
    });

    const modifiedData = Object.assign({}, storedTableState, newColumnState);
    saveStorage(tableIndentifier, modifiedData);
    return modifiedData;
  }

  render() {
    const {
      filterEnabled, columns, data, tableStyle, tableIndentifier, defaultPageSize, ...otherProps
    } = this.props;

    return (
      <div className='table-container'>
        <Table
          data-qa={tableIndentifier}
          style={tableStyle}
          data={data}
          columns={columns}
          onResizedChange={this.handleTableResizeChange}
          defaultPageSize={defaultPageSize}
          pageSizeOptions={[5, 10, 15, 20, 25, 30, 50, 100]}
          resizable
          {...otherProps}
        />
        {filterEnabled && (
          <div className='column-selector'>
            <IconButton
              icon={Filter}
              onClick={this.showColumnSelector}
              title='Filter Columns'
            />
          </div>
        )
        }
      </div>
    );
  }
}

EnhancedTable.defaultProps = {
  defaultPageSize: 10,
  minRows: 0,
  resizable: true,
  tableStyle: {},
  filterEnabled: false,
};

EnhancedTable.propTypes = {
  tableIndentifier: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired,
  columns: PropTypes.array.isRequired,
  defaultPageSize: PropTypes.number,
  minRows: PropTypes.number,
  resizable: PropTypes.bool,
  filterEnabled: PropTypes.bool,
  tableStyle: PropTypes.object,
};
