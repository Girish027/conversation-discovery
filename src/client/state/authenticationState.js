import {
  fromJS, Map
} from 'immutable';
import { createSelector } from 'reselect';
import axios from 'axios'; // this module is Promise based HTTP client for the browser and node.js
import appLogger from 'utils/appLogger';
import { getUserConfig } from 'state/appState';
import constants from '../constants';
import { initializeAnalytics, logAmplitudeEvent } from '../utils/amplitudeUtils';

export const types = {
  AUTHENTICATION_COMPLETE: 'auth/AUTHENTICATION_COMPLETE',
  LOGGED_OUT: 'auth/LOGGED_OUT'
};

export const stateKey = {
  name: 'name',
  oktaUrl: 'oktaUrl',
  email: 'email'
};

const devlog = appLogger({ tag: 'authenticationState' });

export const initialAuthState = fromJS({
  name: 'User Name',
  oktaUrl: ''
});

export const actionCompletedAuth = (userInfo) => ({
  type: types.AUTHENTICATION_COMPLETE,
  userInfo
});

export const actionLoggedoutAuth = () => ({
  type: types.LOGGED_OUT
});

export const getAuthUserinfo = () => (dispatch) => {
  axios.get(constants.serverApiUrls.authentications)
    .then((res) => {
      const authInfoArray = res.data;
      if (Array.isArray(authInfoArray) && authInfoArray.length) {
        const authInfo = authInfoArray[0];
        const userInfo = {
          name: authInfo.name,
          oktaUrl: authInfo.oktaUrl,
          email: authInfo.email,
        };
        initializeAnalytics(authInfo.analyticsKey, authInfo.name);
        const IDTUserEvent = { username: authInfo.name, toolId: constants.toolName, env: constants.environment };
        logAmplitudeEvent('IDTUser', IDTUserEvent);
        dispatch(actionCompletedAuth(userInfo));
        dispatch(getUserConfig());
      } else {
        dispatch(actionLoggedoutAuth());
      }
    })
    .catch((fetchErr) => {
      devlog.error(fetchErr);
    });
};

// REDUCER
export const authReducer = (state = initialAuthState, action = {}) => {
  switch (action.type) {
    case types.AUTHENTICATION_COMPLETE:
      return state.merge(Map(action.userInfo));
    case types.LOGGED_OUT:
      return initialAuthState;
    default:
      return state;
  }
};

// SELECTORS
export const getAuthState = (state) => state.get('auth');

export const getAuthNameSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.name),
);

export const getOktaUrlSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.oktaUrl),
);

export const getEmailSelector = createSelector(
  getAuthState,
  (authState) => authState.get(stateKey.email),
);
