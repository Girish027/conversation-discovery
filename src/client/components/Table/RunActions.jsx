import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Xmark,
  Pencil,
  Download
} from '@tfs/ui-components';
import IconButton from 'components/IconButton';
import {
  setModalIsOpen,
} from 'state/appState';
import Constants from '../../constants';
import { runActions } from 'state/runsState';
import * as globalActions from 'state/globalActions';

class RunActions extends Component {
  constructor() {
    super();

    this.iconProps = {
      padding: '5px',
      height: 12,
      width: 12,
    };
  }

  onClickEdit = (event) => {
    event.stopPropagation();
    const { dispatch, runId, ...runInfo } = this.props;
    dispatch(setModalIsOpen(true, {
      modalName: Constants.modals.editRun,
      runId,
      formData: runInfo,
    }));
  }

  onClickConfirmDelete = () => {
    const { dispatch, runId } = this.props;
    dispatch(runActions.deleteRun(runId));
  }

  onClickDelete = (event) => {
    event.stopPropagation();
    const { dispatch, runName } = this.props;
    dispatch(setModalIsOpen(true, {
      modalName: Constants.modals.confirm,
      deleteMode: true,
      header: `Delete "${runName}"?`,
      message: 'These analysis results will be gone forever. Are you sure you want to delete this item?',
      onClickOk: this.onClickConfirmDelete,
    }));
  }

  onClickDownload = (event) => {
    event.stopPropagation();
    const { dispatch, runId } = this.props;
    dispatch(globalActions.downloadResults(Constants.serverApiUrls.downloadRun, runId));
  }

  renderDeleteIcon = () => {
    const { runStatus } = this.props;
    let disabled = false;
    switch (runStatus) {
      case Constants.status.QUEUED:
      case Constants.status.IN_PROGRESS:
        disabled = true;
        break;
      default:
        break;
    }
    return (
      <IconButton
        onClick={this.onClickDelete}
        icon={Xmark}
        data-qa={'delete-run'}
        title='Delete'
        disabled={disabled}
        {...this.iconProps}
      />
    );
  }

  renderEditIcon = () => (
    <IconButton
      onClick={this.onClickEdit}
      icon={Pencil}
      defaultColor='none'
      strokeColor='#004c97'
      data-qa={'edit-run'}
      title='Edit'
      {...this.iconProps}
    />
  )

  renderDownloadIcon = () => {
    const { runStatus, projectType } = this.props;
    let disabled = true;
    if (runStatus === Constants.status.READY && projectType !== Constants.projects.SYSTEM) {
      disabled = false;
    }
    return (
      <IconButton
        onClick={this.onClickDownload}
        icon={Download}
        data-qa={'delete-run'}
        title='Download Run Results'
        disabled={disabled}
        {...this.iconProps}
      />
    );
  }

  render() {
    const { roleConfig } = this.props;
    return (
      <div className='float-left'>
        { roleConfig.deleteAnalysis ? this.renderDeleteIcon() : null }
        {this.renderDownloadIcon()}
        { roleConfig.updateAnalysis ? this.renderEditIcon() : null }
      </div>
    );
  }
}

RunActions.propTypes = {
  runName: PropTypes.string.isRequired,
  runDescription: PropTypes.string,
  runId: PropTypes.string.isRequired,
  runStatus: PropTypes.string.isRequired,
  dispatch: PropTypes.func,
  projectType: PropTypes.string,
  roleConfig: PropTypes.object,
};

RunActions.defaultProps = {
  runDescription: '',
  roleConfig: {},
};

export default RunActions;
