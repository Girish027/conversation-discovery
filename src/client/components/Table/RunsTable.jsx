import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  StatusBadge, DashedSpinner
} from '@tfs/ui-components';
import RunActions from 'components/Table/RunActions';
import { projectActions } from 'state/projectsState';
import EnhancedTable, { getColumnVisibility, getColumnWidth } from 'components/Table/EnhancedTable';
import { runActions } from 'state/runsState';
import { colors } from 'styles';
import { formatTimestamp } from 'utils/DateUtils';
import { shouldHandle } from 'utils/KeyboardUtils';
import Constants from 'components/../constants';
import GenericTooltip from '../Tooltip/Tooltip';
import { loadStorage } from '../../utils/storageManager';
import constants from '../../constants';

export class RunsTable extends Component {
  constructor(props) {
    super(props);
    this.tableIndentifier = 'CFD_Runs_Table';
    this.tableInfo = {
      name: {
        header: 'Name',
        id: 'runName',
      },
      modified: {
        header: 'Modified on',
        id: 'modified',
      },
      status: {
        header: 'Status',
        id: 'runStatus',
      },
      numOfClusters: {
        header: 'Clusters',
        id: 'numOfClusters',
      },
      numOfTurns: {
        header: 'Turns',
        id: 'numOfTurns',
      },
      modifiedBy: {
        header: 'Modified By',
        id: 'modifiedBy',
      },
      description: {
        header: 'Description',
        id: 'runDescription',
      },
      actions: {
        header: 'Actions',
        id: 'actions'
      }
    };
    this.statusCategoryMapping = {
      QUEUED: 'secondary',
      'IN PROGRESS': 'primary',
      FAILED: 'danger',
      READY: 'primary',
      COMPLETE: 'success',
    };
    this.runsTimer = null;
  }

  componentDidUpdate() {
    const { dispatch } = this.props;
    if (!this.runsTimer) {
      this.runsTimer = setInterval(() => {
        dispatch(runActions.updateRunsStatus());
      }, Constants.pollingInterval);
    }
  }

  componentWillUnmount() {
    clearInterval(this.runsTimer);
    this.runsTimer = null;
  }

  onClickStarRun = ({ runId, starred }) => {
    const { dispatch } = this.props;
    dispatch(runActions.editRun(runId, {
      starred: starred ? 0 : 1
    }, false));
  }

  getType = (runStatus) => (
    runStatus === Constants.status.FAILED
      ? 'error'
      : 'info'
  );

  getClickStyle = (runStatus) => (
    ((runStatus === Constants.status.READY) || (runStatus === Constants.status.COMPLETE))
      ? { pointerEvents: 'auto' }
      : { pointerEvents: 'none' }
  );

  get columns() {
    const { dispatch, projectType, roleConfig } = this.props;
    const tableState = loadStorage(this.tableIndentifier, {});

    const columns = [
      {
        Header: <div className='link-text-run'>{this.tableInfo.name.header}</div>,
        accessor: this.tableInfo.name.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.name.id, 140),
        show: getColumnVisibility(tableState, this.tableInfo.name.id, true),
        Cell: ({ original }) => (
          <div
            className='link-text'
            onClick={() => this.selectRun(original)}
            role='button'
            tabIndex={0}
            onKeyDown={(event) => shouldHandle(event) && this.selectRun(original)}
            style={this.getClickStyle(original.runStatus)}
          >
            {original.runName}
          </div>
        )
      },
      {
        Header: this.tableInfo.modified.header,
        accessor: this.tableInfo.modified.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.modified.id, 110),
        show: getColumnVisibility(tableState, this.tableInfo.modified.id, true),
        Cell: ({ original }) => (
          <div>
            {formatTimestamp(original.modified)}

          </div>
        )
      },
      {
        Header: this.tableInfo.status.header,
        accessor: this.tableInfo.status.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.status.id, 140),
        show: getColumnVisibility(tableState, this.tableInfo.status.id, true),
        Cell: ({ original }) => this.renderRunStatus(original),
      },
      {
        Header: this.tableInfo.numOfClusters.header,
        accessor: this.tableInfo.numOfClusters.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.numOfClusters.id, 100),
        show: getColumnVisibility(tableState, this.tableInfo.numOfClusters.id, true),
        Cell: ({ original }) => (
          <div style={{ margin: '1px 20px 0px 0px' }}>
            { original.numOfClusters >= 0
              ? original.numOfClusters : original.runStatus === 'FAILED' ? 'N/A' : (
                <DashedSpinner width={15} height={15} fill={colors.darkText} />
              )
            }
          </div>
        )
      },
      {
        Header: this.tableInfo.numOfTurns.header,
        accessor: this.tableInfo.numOfTurns.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.numOfTurns.id, 85),
        show: getColumnVisibility(tableState, this.tableInfo.numOfTurns.id, true),
      },
      {
        Header: this.tableInfo.modifiedBy.header,
        accessor: this.tableInfo.modifiedBy.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.modifiedBy.id, 150),
        show: getColumnVisibility(tableState, this.tableInfo.modifiedBy.id, true),
      },
      {
        Header: this.tableInfo.description.header,
        accessor: this.tableInfo.description.id,
        sortable: true,
        minWidth: getColumnWidth(tableState, this.tableInfo.description.id, 120),
        show: getColumnVisibility(tableState, this.tableInfo.description.id, true),
      },
      {
        Header: this.tableInfo.actions.header,
        accessor: this.tableInfo.actions.id,
        sortable: false,
        minWidth: getColumnWidth(tableState, this.tableInfo.actions.id, 110),
        Cell: ({ original }) => (
          <RunActions {...original} dispatch={dispatch} projectType={projectType} roleConfig={roleConfig} />
        ),
      },
    ];

    return columns;
  }

  renderRunStatus = ({ runStatus, runStatusDescription }) => {
    const progress = (runStatus === Constants.status.IN_PROGRESS || runStatus === Constants.status.VERIFYING)
      ? {
        type: 'spinner',
      }
      : {};
    return (
      <GenericTooltip content={runStatusDescription} type={this.getType(runStatus)}>
        <div style={{ margin: '0px -5px' }}>
          <StatusBadge
            category={this.statusCategoryMapping[runStatus]}
            label={runStatus}
            onHover={() => { }}
            progress={progress}
            type='badge'
          />
        </div>
      </GenericTooltip>
    );
  }

  selectRun = (run) => {
    const { dispatch, projectId, projectName } = this.props;
    if (projectName === constants.LIVE_DATA_ANALYSIS) {
      dispatch(projectActions.editProject(
        projectId,
        {
          projectName: constants.LIVE_DATA_IN_REVIEW,
          projectStatus: constants.status.IN_REVIEW,
        },
        true,
        run.runId
      ));
    } else {
      dispatch(runActions.selectRun(run.runId));
    }
  }

  render() {
    const { data } = this.props;
    return (
      <EnhancedTable
        tableIndentifier={this.tableIndentifier}
        columns={this.columns}
        data={data}
        tableStyle={{ maxHeight: 'calc(95vh - 150px)' }}
      />
    );
  }
}

RunsTable.propTypes = {
  dispatch: PropTypes.func,
  data: PropTypes.array,
  projectId: PropTypes.string,
  projectName: PropTypes.string,
  projectType: PropTypes.string,
  roleConfig: PropTypes.object,
};

RunsTable.defaultProps = {
  dispatch: () => { },
  data: [],
  projectId: '',
  projectName: '',
  projectType: PropTypes.string,
  roleConfig: {},
};

const mapDispatchToProps = (dispatch) => ({
  dispatch,
});

export default connect(mapDispatchToProps)(RunsTable);
