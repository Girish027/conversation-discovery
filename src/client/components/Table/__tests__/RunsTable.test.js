import React from 'react';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { RunsTable } from 'components/Table/RunsTable';
import { runActions } from 'state/runsState';

describe('<RunsTable />', function () {
  let wrapper;

  const data = [{
    runId: 'run-59da2519-ed17-46ad-8294-1367e287589d',
    projectId: 'pro-be2a948f-3a0d-4c30-fd6e-5157f9278bc2',
    runName: 'test-1564166348646',
    runDescription: 'trial',
    numOfTurns: 4,
    numOfClusters: 400,
    stopWords: '["hello","hilton"]',
    modified: 1564166848680,
    modifiedBy: 'abc@test.com',
    created: 1564166348680,
    createdBy: 'abc@test.com',
    runStatus: 'QUEUED',
    runStatusDescription: 'The run is queued',
    starred: 0
  }, {
    runId: 'run-9dd719a7-4670-4198-5bde-82477be2c5a5',
    projectId: 'pro-be2a948f-3a0d-4c30-fd6e-5157f9278bc2',
    runName: 'test-1564159900138',
    runDescription: 'trial',
    numOfTurns: 4,
    numOfClusters: 400,
    stopWords: '["hello","hilton"]',
    modified: 1564159930153,
    modifiedBy: 'abc@test.com',
    created: 1564159900153,
    createdBy: 'abc@test.com',
    runStatus: 'QUEUED',
    runStatusDescription: 'The run is queued',
    starred: 1
  }];


  const props = {
    data,
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<RunsTable
    {...propsObj}
  />);

  const getMountedWrapper = (propsObj) => mount(<RunsTable
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getMountedWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for the given props', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    ['QUEUED', 'IN PROGRESS', 'FAILED', 'READY'].forEach((runStatus) => {
      test('renders the status in the table ', () => {
        wrapper = getShallowWrapper(props);
        const statusWrapper = shallow(
          <div>
            {' '}
            { wrapper.instance().renderRunStatus({ runStatus, runStatusDescription: 'desc' })}
            {' '}
          </div>
        );
        expect(toJSON(statusWrapper)).toMatchSnapshot();
      });
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.dispatch = jest.fn();
      runActions.selectRun = jest.fn(() => 'selectRun');
      wrapper = getShallowWrapper(props);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('selectRun', () => {
      test('should dispatch an action to select the given run', () => {
        const run = { runId: 'run-123' };
        wrapper.instance().selectRun(run);
        expect(runActions.selectRun).toHaveBeenCalledWith(run.runId);
        expect(props.dispatch).toHaveBeenCalledWith('selectRun');
      });
    });
  });
});
