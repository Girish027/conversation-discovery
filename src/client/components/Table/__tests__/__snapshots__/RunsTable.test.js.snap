// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RunsTable /> Snapshots renders correctly for the given props 1`] = `
<EnhancedTable
  columns={
    Array [
      Object {
        "Cell": [Function],
        "Header": <div
          className="link-text-run"
        >
          Name
        </div>,
        "accessor": "runName",
        "minWidth": 140,
        "show": true,
        "sortable": true,
      },
      Object {
        "Cell": [Function],
        "Header": "Modified on",
        "accessor": "modified",
        "minWidth": 110,
        "show": true,
        "sortable": true,
      },
      Object {
        "Cell": [Function],
        "Header": "Status",
        "accessor": "runStatus",
        "minWidth": 140,
        "show": true,
        "sortable": true,
      },
      Object {
        "Cell": [Function],
        "Header": "Clusters",
        "accessor": "numOfClusters",
        "minWidth": 100,
        "show": true,
        "sortable": true,
      },
      Object {
        "Header": "Turns",
        "accessor": "numOfTurns",
        "minWidth": 85,
        "show": true,
        "sortable": true,
      },
      Object {
        "Header": "Modified By",
        "accessor": "modifiedBy",
        "minWidth": 150,
        "show": true,
        "sortable": true,
      },
      Object {
        "Header": "Description",
        "accessor": "runDescription",
        "minWidth": 120,
        "show": true,
        "sortable": true,
      },
      Object {
        "Cell": [Function],
        "Header": "Actions",
        "accessor": "actions",
        "minWidth": 110,
        "sortable": false,
      },
    ]
  }
  data={
    Array [
      Object {
        "created": 1564166348680,
        "createdBy": "abc@test.com",
        "modified": 1564166848680,
        "modifiedBy": "abc@test.com",
        "numOfClusters": 400,
        "numOfTurns": 4,
        "projectId": "pro-be2a948f-3a0d-4c30-fd6e-5157f9278bc2",
        "runDescription": "trial",
        "runId": "run-59da2519-ed17-46ad-8294-1367e287589d",
        "runName": "test-1564166348646",
        "runStatus": "QUEUED",
        "runStatusDescription": "The run is queued",
        "starred": 0,
        "stopWords": "[\\"hello\\",\\"hilton\\"]",
      },
      Object {
        "created": 1564159900153,
        "createdBy": "abc@test.com",
        "modified": 1564159930153,
        "modifiedBy": "abc@test.com",
        "numOfClusters": 400,
        "numOfTurns": 4,
        "projectId": "pro-be2a948f-3a0d-4c30-fd6e-5157f9278bc2",
        "runDescription": "trial",
        "runId": "run-9dd719a7-4670-4198-5bde-82477be2c5a5",
        "runName": "test-1564159900138",
        "runStatus": "QUEUED",
        "runStatusDescription": "The run is queued",
        "starred": 1,
        "stopWords": "[\\"hello\\",\\"hilton\\"]",
      },
    ]
  }
  defaultPageSize={10}
  filterEnabled={false}
  minRows={0}
  resizable={true}
  tableIndentifier="CFD_Runs_Table"
  tableStyle={
    Object {
      "maxHeight": "calc(95vh - 150px)",
    }
  }
/>
`;

exports[`<RunsTable /> Snapshots renders the status in the table  1`] = `
<div>
   
  <GenericTooltip
    content="desc"
    direction="bottom"
    type="info"
  >
    <div
      style={
        Object {
          "margin": "0px -5px",
        }
      }
    >
      <withTheme(StatusBadge)
        category="secondary"
        label="QUEUED"
        onHover={[Function]}
        progress={Object {}}
        type="badge"
      />
    </div>
  </GenericTooltip>
   
</div>
`;

exports[`<RunsTable /> Snapshots renders the status in the table  2`] = `
<div>
   
  <GenericTooltip
    content="desc"
    direction="bottom"
    type="info"
  >
    <div
      style={
        Object {
          "margin": "0px -5px",
        }
      }
    >
      <withTheme(StatusBadge)
        category="primary"
        label="IN PROGRESS"
        onHover={[Function]}
        progress={
          Object {
            "type": "spinner",
          }
        }
        type="badge"
      />
    </div>
  </GenericTooltip>
   
</div>
`;

exports[`<RunsTable /> Snapshots renders the status in the table  3`] = `
<div>
   
  <GenericTooltip
    content="desc"
    direction="bottom"
    type="error"
  >
    <div
      style={
        Object {
          "margin": "0px -5px",
        }
      }
    >
      <withTheme(StatusBadge)
        category="danger"
        label="FAILED"
        onHover={[Function]}
        progress={Object {}}
        type="badge"
      />
    </div>
  </GenericTooltip>
   
</div>
`;

exports[`<RunsTable /> Snapshots renders the status in the table  4`] = `
<div>
   
  <GenericTooltip
    content="desc"
    direction="bottom"
    type="info"
  >
    <div
      style={
        Object {
          "margin": "0px -5px",
        }
      }
    >
      <withTheme(StatusBadge)
        category="primary"
        label="READY"
        onHover={[Function]}
        progress={Object {}}
        type="badge"
      />
    </div>
  </GenericTooltip>
   
</div>
`;
